docs/communicate-with-javascript.md

```diff
@@ -3154,3 +3165,622 @@ let default = 10;
+end
+
+let foo = Foo.make ()
+let () = Foo.meth foo
+```
+```reasonml
+module Foo = {
+  type t;
+
+  [@mel.new] external make: unit => t = "Foo";
+  [@mel.send] external meth: t => unit = "meth";
+};
+
+let foo = Foo.make();
+let () = Foo.meth(foo);
+```
+
+See the [Calling an object method](#calling-an-object-method) section for more
+information.
+
+#### `const newStr = str.replace(substr, newSubstr)`: non-mutating instance method
+
+```ocaml
+external replace : substr:string -> newSubstr:string -> string = "replace"
+[@@mel.send.pipe: string]
+
+let str = "goodbye world"
+let substr = "goodbye"
+let newSubstr = "hello"
+let newStr = replace ~substr ~newSubstr str
+```
+```reasonml
+[@mel.send.pipe: string]
+external replace: (~substr: string, ~newSubstr: string) => string = "replace";
+
+let str = "goodbye world";
+let substr = "goodbye";
+let newSubstr = "hello";
+let newStr = replace(~substr, ~newSubstr, str);
+```
+
+`mel.send.pipe` injects a parameter of the given type (in this case `string`) as
+the final positional parameter of the binding. In other words, it creates the
+binding with the real signature <code class="text-ocaml">substr:string -\>
+newSubstr:string -\> string -\> string</code><code
+class="text-reasonml">(~substr: string, ~newSubstr: string, string) =\>
+string</code>. This is handy for non-mutating functions as they traditionally
+take the instance as the final parameter.
+
+It is not strictly necessary to use named arguments in this binding, but it
+helps readability with multiple arguments, especially if some have the same
+type.
+
+Also note that it is not strictly need to use `mel.send.pipe`, one can use
+`mel.send` everywhere.
+
+See the [Calling an object method](#calling-an-object-method) section for more
+information.
+
+#### `arr.sort(compareFunction)`: mutating instance method
+
+<!--#prelude#let arr = [|2|]-->
+```ocaml
+external sort : 'a array -> (('a -> 'a -> int)[@mel.uncurry]) -> 'a array
+  = "sort"
+[@@mel.send]
+
+let _ = sort arr compare
+```
+```reasonml
+[@mel.send]
+external sort: (array('a), [@mel.uncurry] (('a, 'a) => int)) => array('a) =
+  "sort";
+
+let _ = sort(arr, compare);
+```
+
+For a mutating method, it's traditional to pass the instance argument first.
+
+Note: `compare` is a function provided by the standard library, which fits the
+defined interface of JavaScript's comparator function.
+
+### Null and undefined
+
+#### `foo.bar === undefined`: check for undefined
+
+```ocaml
+module Foo = struct
+  type t
+  external bar : t -> int option = "bar" [@@mel.get]
+end
+
+external foo : Foo.t = "foo"
+
+let _result = match Foo.bar foo with Some _ -> 1 | None -> 0
+```
+```reasonml
+module Foo = {
+  type t;
+  [@mel.get] external bar: t => option(int) = "bar";
+};
+
+external foo: Foo.t = "foo";
+
+let _result =
+  switch (Foo.bar(foo)) {
+  | Some(_) => 1
+  | None => 0
+  };
+```
+
+If you know some value may be `undefined` (but not `null`, see next section),
```

Comment:
```suggestion
If you know some value may be `undefined` (but not `null`), see next section,
```

Reply:
I believe this change modifies the meaning. The original meaning was that if the value may be `null`, then the reader should see next section. After this change that meaning is lost.

Reply:
Perhaps
```
If you know some value may be `undefined` (but not `null`, see the next section for more on `null`), and if you know...
```